File: backend/ResearchCruiseApp/Application/Models/DTOs/CruiseApplications/ContractDto.cs
namespace ResearchCruiseApp.Application.Models.DTOs.CruiseApplications;

public class ContractDto
{
    public string Category { get; init; } = null!;

    public string? InstitutionName { get; init; }

    public string? InstitutionUnit { get; init; }

    public string? InstitutionLocalization { get; init; }

    public string? Description { get; init; }

    public List<FileDto> Scans { get; set; } = [];
}

File: frontend/src/modules/cruise-applications/models/ContractDto.ts
import { z } from 'zod';

import { FileDto, FileDtoValidationSchema } from '@/cruise-applications/models/FileDto';

export type ContractDto = {
  category: 'domestic' | 'international';
  institutionName: string;
  institutionUnit: string;
  institutionLocalization: string;
  description: string;
  scans: FileDto[];
};

export function getContractCategoryName(category: ContractDto['category']): string {
  return category === 'domestic' ? 'Krajowa' : 'Międzynarodowa';
}

export const ContractDtoValidationSchema = z.object({
  category: z.enum(['domestic', 'international']),
  institutionName: z.string().nonempty('Nazwa instytucji jest wymagana'),
  institutionUnit: z.string().nonempty('Jednostka jest wymagana'),
  institutionLocalization: z.string().nonempty('Lokalizacja instytucji jest wymagana'),
  description: z.string().nonempty('Opis jest wymagany').max(10240, 'Maksymalna długość to 10240 znaków'),
  scans: FileDtoValidationSchema.array(),
});
